server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/materials/modals.tsx

```diff
@@ -51,3 +51,37 @@ export class AddMaterialModal extends Modal {
     }
   }
 }
+
+export class EditMaterialModal extends Modal {
+  private readonly material: Material;
+  private readonly onSuccessfulEdit: (oldMaterial: Material, newMaterial: Material) => void;
+  private readonly _oldMaterial: Material;
+
+  constructor(material: Material, onSuccessfulEdit: (oldMaterial: Material, newMaterial: Material) => void) {
+    super(Size.large);
+    this._oldMaterial     = material;
+    this.material         = material.clone();
```

Comment:
But `_.cloneDeep()` does not work because values are in the form of `stream`. Did you try editing material and close modal instead of clicking on the update button? In this case, material on index page should not get updated.

Reply:
ok, got it 👍